### PR TITLE
Remove unused 'remove_former_image' parameter in squash

### PR DESCRIPTION
**Pull request by rcerven**
remove_former_image argument isn't used anymore at all

### CODE REVIEW

File: atomic_reactor/plugins/prepub_squash.py [hunk 1]
```diff
@@ -57,14 +55,13 @@ class PrePublishSquashPlugin(PrePublishPlugin):
     is_allowed_to_fail = False
 
     def __init__(self, tasker, workflow, tag=None, from_base=True, from_layer=None,
-                 remove_former_image=True, dont_load=False):
+                 dont_load=False):
```
Comment (twaugh) [inline atomic_reactor/plugins/prepub_squash.py]: I don't think we can just remove it from the parameter list -- older osbs-clients will generate build JSON that will no longer work.

Comment (rcerven) [inline atomic_reactor/plugins/prepub_squash.py]: so should I just leave it in args and in docstring write : not used ? something like that?

Comment (twaugh) [inline atomic_reactor/plugins/prepub_squash.py]: Yes, I think so.

This is not actually set by default in osbs-client, but we should leave a grace period and mention it in the release notes when we release atomic-reactor next.

Comment (rcerven) [inline atomic_reactor/plugins/prepub_squash.py]: ok

**Comment by rcerven** [comment]
squashed

**Comment by twaugh** [comment]
Maybe a better commit message would be "Mark squash plugin 'remove_former_image' parameter as not used".

File: atomic_reactor/plugins/prepub_squash.py [hunk 2]
```diff
@@ -64,7 +62,7 @@ def __init__(self, tasker, workflow, tag=None, from_base=True, from_layer=None,
         :param from_base: bool, squash from base-image layer, on by default
         :param from_layer: layer from we will squash - if specified, takes precedence over from_base
         :param tag: str, new name of the image - by default use the former one
-        :param remove_former_image: bool, remove unsquashed image?
+        :param remove_former_image: not used
```
Comment (lcarva) [inline atomic_reactor/plugins/prepub_squash.py]: Can this actually be removed from the constructor? `atomic-reactor` should ignore unknown args: https://github.com/projectatomic/atomic-reactor/blob/b14b5abd75e2b1cb67be6e39fafcf2b98e5a8bbe/atomic_reactor/plugin.py#L326

Comment (twaugh) [inline atomic_reactor/plugins/prepub_squash.py]: Oh! I hadn't realised it did that... :(

Comment (rcerven) [inline atomic_reactor/plugins/prepub_squash.py]: so you mean remove that docstring about that param?

Comment (twaugh) [inline atomic_reactor/plugins/prepub_squash.py]: You can remove that parameter from the docstring and also from the constructor's signature... like you had before I asked you to change it (sorry).

Comment (rcerven) [inline atomic_reactor/plugins/prepub_squash.py]: ahhh ok

**Comment by twaugh** [comment]
The commit history could have been cleaned up before this got merged :(

**Comment by lcarva** [comment]
I squashed the commits prior to merging it.

**Pull request merged**